lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp

```diff
@@ -81,6 +81,31 @@ class TosaToTTIRMultiplyOpConversionPattern
   }
 };
 
+template <typename SrcOp, typename DestOp,
+          typename Adaptor = typename SrcOp::Adaptor>
+class TosaToTTIRReduceOpConversionPattern : public OpConversionPattern<SrcOp> {
+  using OpConversionPattern<SrcOp>::OpConversionPattern;
+
+public:
+  LogicalResult
+  matchAndRewrite(SrcOp srcOp, Adaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto outputType = mlir::cast<RankedTensorType>(srcOp.getResult().getType());
+    auto outputTensor = rewriter.create<tensor::EmptyOp>(
```

Comment:
Forgot to comment, let's try avoiding using auto whenever we can, except when type is very complicated. You can use `RankedTensorType` and `tensor::EmptyOp` for these two autos.

Reply:
fixed